kubectl-cloudflow/cmd/verify.go

```diff
@@ -0,0 +1,143 @@
+			config, err := loadBlueprintConfigContent(blueprint)
+			if err != nil {
+				util.LogAndExit("failed to parse the blueprint contents")
+			}
+			println(config.String())
+			err = verify.VerifyBlueprint(config)
+			if err != nil {
+				util.LogAndExit("Blueprint verification failed. Error: %s", err.Error())
+			} else {
+				print("Blueprint verification succeeded.")
+			}
+
+		} else { // this is a normal url
+		    tempDir, err := getTempDir()
+
+		    if err!= nil {
+		    	util.LogAndExit("Local temp dir not properly defined. Error: %s", err.Error())
+			}
+			resp, err := client.Get(blueprint)
+
+			if err != nil {
+				util.LogAndExit("You need to specify the full path to a blueprint file. %s, is not accessible. Error: %s ", blueprint, err.Error())
+			}
+
+			if resp == nil || resp.StatusCode != 200 {
+				util.LogAndExit("You need to specify the full path to a blueprint file. %s, is not accessible", blueprint)
+			}
+
+			file, err := ioutil.TempFile(tempDir, path.Base(blueprintURL.Path)+".")
+			if err != nil {
+				util.LogAndExit("Could not create a temp file. Error: %s",  err.Error())
+			}
+
+			if _, err = io.Copy(file, resp.Body); err != nil {
+				util.LogAndExit("Could not copy the remote blueprint file.  Error: %s",  err.Error())
+			}
+
+			if err = file.Close(); err != nil {
+				util.LogAndExit("Failed to close the file.  Error: %s",  err.Error())
+			}
+			config, _ := loadBlueprintConfigContent(file.Name())
+			println(config.String())
+			err = verify.VerifyBlueprint(config)
+			if err != nil {
+				util.LogAndExit("Blueprint verification failed. Error: %s", err.Error())
+			} else {
+				print("Blueprint verification succeeded.")
+			}
+		}
+	}
+}
+
+func getTempDir() (string, error) {
+	tempDir, isPresent := os.LookupEnv("CLOUDFLOW_BLUEPRINT_DOWNLOAD_DIR")
+	if !isPresent {
+		return os.TempDir(), nil
+	} else {
+		if len(tempDir) == 0 {
+			return "", fmt.Errorf("You need to specify a non-empty CLOUDFLOW_BLUEPRINT_DOWNLOAD_DIR env var")
+		} else {
+			return tempDir, nil
+		}
+	}
+}
+
+func validateVerifyCmdArgs(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("you need to specify the full path to a blueprint file")
+	}
+	blueprint:= args[0]
+	blueprintURL, err := url.Parse(blueprint)
+
+	if err != nil {
+		return fmt.Errorf("You need to specify the full path to a blueprint file. %s, is malformed.", blueprint)
+	}
+
+	if blueprintURL.Scheme == "" {
+		if !util.FileExists(blueprint) {
+			return fmt.Errorf("You need to specify the full path to a blueprint file. %s, local file does not exist.", blueprint)
+		}
+	}
+	return nil
+}
+
+func loadBlueprintConfigContent(path string) (*configuration.Config, error) {
+	var sb strings.Builder
+	println(path)
+	content, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("could not read configuration file %s", content)
+	}
+	sb.Write(content)
+	sb.WriteString("")
+	config := configuration.ParseString(sb.String())
```

Comment:
@skonto Does this `configuration.ParseString(..)` work when you have images with tags ? Can u pls try if the following parses into a valid HOCON config .. I suspect that the ":" character creates problems in parsing ..

```
blueprint {
  images {
    proc = eu.gcr.io/bubbly-observer-178213/processor:34-69082eb-dirty
    ings = eu.gcr.io/bubbly-observer-178213/ingestor:34-69082eb-dirty
    logr = eu.gcr.io/bubbly-observer-178213/logger:34-69082eb-dirty
  }
  streamlets {
    taxi-ride = ings/taxiride.ingestor.TaxiRideIngress
    taxi-fare = ings/taxiride.ingestor.TaxiFareIngress
    processor = proc/taxiride.processor.TaxiRideProcessor
    logger = logr/taxiride.logger.FarePerRideLogger
  }
  connections {
    taxi-ride.out = [processor.in-taxiride]
    taxi-fare.out = [processor.in-taxifare]
    processor.out = [logger.in]
  }
}
```

Reply:
According to the hocon spec https://github.com/lightbend/config/blob/master/HOCON.md#unquoted-strings: "A sequence of characters outside of a quoted string is a string value if:

it does not contain "forbidden characters": '$', '"', '{', '}', '[', ']', ':', '=', ',', '+', '#', '`', '^', '?', '!', '@', '*', '&', '' (backslash), or whitespace.
it does not contain the two-character string "//" (which starts a comment)
its initial characters do not parse as true, false, null, or a number."
So we need to use quotes.